### ISSUE TITLE
Trouble building with gcc 14.2.0

### ISSUE THREAD

**Issue by jcurtiss8086** [suzizecat/slang-lsp-tools#14]
I'm trying to build the latest commit in the repo on SLES15 using GCC 14.2.0 and am getting the following compile error.   My cmake commands are identical to the README except I added -DCMAKE_CXX_COMPILER=/path/to/gcc/14.2.0/bin/g++

Have you run into this kind of problem before?

```
In file included from slang-lsp-tools/lsp-server/diplomat/include/diplomat_lsp.hpp:8,
                 from slang-lsp-tools/lsp-server/diplomat/src/diplomat_lsp.cpp:1:
slang-lsp-tools/lsp-server/diplomat/include/diplomat_index.hpp:9:13: error: redefinition of ‘struct std::hash<slang::SourceRange>’
    9 | struct std::hash<slang::SourceRange>
      |             ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from slang-lsp-tools/build/_deps/slang-src/source/../include/slang/diagnostics/Diagnostics.h:18,
                 from slang-lsp-tools/build/_deps/slang-src/source/../include/slang/syntax/SyntaxTree.h:13,
                 from slang-lsp-tools/lsp-server/diplomat/include/sv_document.hpp:7,
                 from slang-lsp-tools/lsp-server/diplomat/include/diplomat_lsp.hpp:5:
slang-lsp-tools/build/_deps/slang-src/source/../include/slang/text/SourceLocation.h:241:8: note: previous definition of ‘struct std::hash<slang::SourceRange>’
  241 | struct hash<slang::SourceRange> {
      |        ^~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by suzizecat** [comment]
Hi @jcurtiss8086 

This one is probably on me as I updated the version of slang, forgetting that it included a fix I worded around previously. 

As a workaround, until I fix it, you may set `v6.0` as a git tag for slang in `third-parties.cmake`, remove the `build` directory and reconfigure. It should be working.

**Comment by suzizecat** [comment]
Reopening to keep track and ensure that I don't forget to check that it is fixed
